tests/release_candidate_scraper/test_release_candidate_scraper_latest.py

```diff
@@ -3,202 +3,105 @@
+    ({'application': 'thunderbird', 'platform': 'linux64', 'version': 'latest-beta'},
+     'thunderbird-20.0b1-build1.en-US.linux64.tar.bz2',
+     'thunderbird/candidates/20.0b1-candidates/build1/linux-x86_64/en-US/thunderbird-20.0b1.tar.bz2'),
+    ({'application': 'thunderbird', 'platform': 'mac', 'version': 'latest-beta'},
+     'thunderbird-20.0b1-build1.en-US.mac.dmg',
+     'thunderbird/candidates/20.0b1-candidates/build1/mac/en-US/Thunderbird 20.0b1.dmg'),
+    ({'application': 'thunderbird', 'platform': 'win32', 'version': 'latest-beta'},
+     'thunderbird-20.0b1-build1.en-US.win32.exe',
+     'thunderbird/candidates/20.0b1-candidates/build1/win32/en-US/Thunderbird Setup 20.0b1.exe'),
+    ({'application': 'thunderbird', 'platform': 'linux', 'version': 'latest-esr'},
+     'thunderbird-17.0.1esr-build1.en-US.linux.tar.bz2',
+     'thunderbird/candidates/17.0.1esr-candidates/build1/linux-i686/en-US/thunderbird-17.0.1esr.tar.bz2'),
+    ({'application': 'thunderbird', 'platform': 'linux64', 'version': 'latest-esr'},
+     'thunderbird-17.0.1esr-build1.en-US.linux64.tar.bz2',
+     'thunderbird/candidates/17.0.1esr-candidates/build1/linux-x86_64/en-US/thunderbird-17.0.1esr.tar.bz2'),
+    ({'application': 'thunderbird', 'platform': 'mac', 'version': 'latest-esr'},
+     'thunderbird-17.0.1esr-build1.en-US.mac.dmg',
+     'thunderbird/candidates/17.0.1esr-candidates/build1/mac/en-US/Thunderbird 17.0.1esr.dmg'),
+    ({'application': 'thunderbird', 'platform': 'win32', 'version': 'latest-esr'},
+     'thunderbird-17.0.1esr-build1.en-US.win32.exe',
+     'thunderbird/candidates/17.0.1esr-candidates/build1/win32/en-US/Thunderbird Setup 17.0.1esr.exe'),
 
+])
 
-class ReleaseCandidateScraperTest(mhttpd.MozHttpdBaseTest):
-    """test mozdownload scraper class"""
 
-    def test_latest_build(self):
-        """Testing various download scenarios for latest release candidate builds"""
+def test_latest_build(httpd, tmpdir, args, filename, url):
+    """Testing various download scenarios for latest release candidate builds"""
 
-        for entry in tests:
-            scraper = ReleaseCandidateScraper(destination=self.temp_dir,
-                                              base_url=self.wdir,
-                                              logger=self.logger,
-                                              **entry['args'])
-            expected_filename = os.path.join(self.temp_dir, entry['filename'])
-            self.assertEqual(scraper.filename, expected_filename)
-            self.assertEqual(urllib.unquote(scraper.url),
-                             urljoin(self.wdir, entry['url']))
+    scraper = ReleaseCandidateScraper(destination=str(tmpdir), base_url=httpd.get_url(), **args)
+    expected_filename = os.path.join(str(tmpdir), filename)
+    assert urllib.unquote(scraper.url) == urljoin(httpd.get_url(), url)
+    assert scraper.filename == expected_filename
```

Comment:
Switch it with the line above, so that it is close to the definition of `expected_filename`. Both belong together.